quote-book/src/synchronized_quote_book.rs

```diff
@@ -3,27 +3,100 @@
 use crate::{Error as QuoteBookError, Pair, Quote, QuoteBook, QuoteId};
 use mc_blockchain_types::BlockIndex;
 use mc_crypto_ring_signature::KeyImage;
-use mc_ledger_db::Ledger;
+use mc_ledger_db::{Error as LedgerError, Ledger};
 use mc_transaction_extra::SignedContingentInput;
-use std::ops::RangeBounds;
 
+use std::{
+    ops::RangeBounds,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc, Mutex,
+    },
+    thread::{Builder as ThreadBuilder, JoinHandle},
+    time::Duration,
+};
+
+use crate::Msg;
+use mc_common::logger::{log, Logger};
+use postage::{broadcast::Sender, prelude::Sink};
 /// A wrapper for a quote book implementation that syncs quotes with the ledger
-#[derive(Clone)]
-pub struct SynchronizedQuoteBook<Q: QuoteBook, L: Ledger + Clone + 'static> {
+pub struct SynchronizedQuoteBook<Q: QuoteBook, L: Ledger + Clone + Sync + 'static> {
     /// Quotebook being synchronized to the ledger by the Synchronized Quotebook
     quote_book: Q,
 
+    /// Ledger
     ledger: L,
+
+    /// Shared state
+    shared_state: Arc<Mutex<DbPollSharedState>>,
```

Comment:
This can be simplified into an atomic value since there is no other shared state